modules/core/src/lib/layer.js

```diff
@@ -679,23 +699,21 @@ export default class Layer extends Component {
 
   // Calculates uniforms
   drawLayer({moduleParameters = null, uniforms = {}, parameters = {}}) {
+    const currentProps = this.props;
     if (!uniforms.picking_uActive) {
-      this.updateTransition();
+      // Overwrite this.props during redraw to use in-transition prop values
+      this.props = this.updateTransition();
     }
 
+    const {opacity} = this.props;
+    // apply gamma to opacity to make it visually "linear"
```

Comment:
Nit: Seems like this gamma conversion should be specified in a table somewhere, rather than being a special case here, could even be in the prop types...